src/background/browser-permissions-tracker.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Interpreter } from 'background/interpreter';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Logger } from 'common/logging/logger';
+import { Message } from 'common/message';
+import { Messages } from 'common/messages';
+
+export const allUrlAndFilePermissions: string = '*://*/*';
+
+export class BrowserPermissionsTracker {
+    constructor(
+        private browserAdapter: BrowserAdapter,
+        private interpreter: Interpreter,
+        private readonly logger: Logger,
+    ) {}
+
+    public async initialize(): Promise<void> {
+        this.browserAdapter.addListenerOnPermissionsAdded(this.notifyChange);
+        this.browserAdapter.addListenerOnPermissionsRemoved(this.notifyChange);
+        await this.notifyChange();
+    }
+
+    private notifyChange = async (): Promise<void> => {
+        let payload: boolean;
+
+        try {
+            payload = await this.browserAdapter.containsPermissions({
+                origins: [allUrlAndFilePermissions],
+            });
+        } catch (error) {
+            payload = false;
+            this.logger.log('Error occurred while checking browser permissions');
+        } finally {
+            const message: Message = {
+                messageType: Messages.PermissionsState.SetPermissionsState,
+                payload: payload,
+                tabId: null,
```

Comment:
Does tabId need to be set as null? Can it be left out entirely?

Reply:
You're right; I took out the tabId.